### DISCUSSION TITLE
Failing to setup local HTTPS

### DISCUSSION THREAD

**Discussion by dakotahorstman** [cookiecutter/cookiecutter-django#4353] [Q&A]
I'm attempting to setup local HTTPS as described in the documentation through docker. I created my `*.pem` files with mkcert and moved them to a `certs/` directory in the root project.

After stepping through the rest of the configuration, I'm unable to reach `https://automation.local/`, which is the URL my server is at. I can reach it at `http://automation.local/`, but not `https`. Chrome shows the error `ERR_CONNECTION_CLOSED` and then `ERR_CONNECTION_RESET`.

I've already edited `/etc/hosts` to include `automation.local     127.0.0.1`, else I wouldn't be able to reach the server at `http`.

Any ideas as to why this is?

**Comment by dakotahorstman** [comment]
Okay, after some further trial and error, I've discovered the Nginx dislikes the `.pem` files from `mkcert`. Using Let's Encrypt's method at least allows me to connect to the server--albeit with an "invalid certificate" warning.

I'm wondering if there is further setup that is needed that is not discussed in the documentation for `mkcert`?

**Comment by ddkasa** [comment]
@dakotahorstman  I am current having similar problem at the moment. Did you manage to full resolve this issue in the end?

**Reply by dakotahorstman** [reply]
I did.

1. Generate the certs using mkcert like normal: e.g. `mkcert automation.local`
2. Place the generated `<domain>-key.pem` and `<domain>.pem` into a `certs/` directory in your project
3. Rename `<domain>-key.pem` to `<domain>.key` and `<domain>.pem` to `<domain>.crt`: e.g. `automation.local-key.pem` -> `automation.local.key` and `automation.local.pem` -> `automation.local.crt`
4. Next, place the following into `local.yml` -- if you have an existing nginx config, remove it.

```docker
nginx:
    image: jwilder/nginx-proxy:latest
    container_name: automation_dev_nginx
    ports:
      - '80:80'
      - '443:443'
    volumes:
      - /var/run/docker.sock:/tmp/docker.sock:ro
      - ./certs:/etc/nginx/certs:ro
    depends_on:
      - django
    restart: always
```
5. Ensure `VIRUTAL_HOST` contains your domain:
**.envs/.local/.django**
```
...

# HTTPS
# ----------------------------------------------------------------------
VIRTUAL_HOST=automation.local
VIRTUAL_PORT=8000
```

6. Profit.

If you encounter troubles, let me know :)
